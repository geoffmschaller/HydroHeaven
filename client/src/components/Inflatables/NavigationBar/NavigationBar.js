import React from 'react';
import {Link} from 'react-router-dom';
import styles from './NavigationBar.module.css';
import MAIN_LOGO from '../../../static/images/main_logo_desktop_no_border.png';

export default class NavigationBar extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			drawerOpen: false,
		};
		this.setSliderDrawer = (mode) => {
			this.setState({drawerOpen: mode});
			if (!mode) {
				console.log("AUTO");
				document.body.style.overflow = 'auto';
			} else {
				console.log("HIDDEN");
				document.body.style.overflow = 'hidden';
			}
		};
	}


	render() {
		const sliderClasses = this.state.drawerOpen ? styles.slider : [styles.slider, styles.closedSlider].join(' ');

		return (
			<>
				<div className={styles.navigationBar}>
					<div className={styles.desktop}>
						<Link to="/service-contact">
							<div className={styles.menuItem}>
								<i className="fad fa-tools"/>
								SERVICE & CONTACT
							</div>
						</Link>
						<Link to="/bbq-islands">
							<div className={styles.menuItem}>
								<i className="fad fa-fire"/>
								BBQs
							</div>
						</Link>
						<Link to="/swim-spas">
							<div className={styles.menuItem}>
								<i className="fad fa-swimmer"/>
								SWIM SPAS
							</div>
						</Link>
						<Link to="/spas-hot-tubs">
							<div className={styles.menuItem}>
								<i className="fad fa-hot-tub"/>
								HOT TUBS & SPAS
							</div>
						</Link>
						<Link to="/">
							<div className={styles.menuItem}>
								<i className="fad fa-home-alt"/>
								HOME
							</div>
						</Link>
						<div className="clear"/>
					</div>
					<div className={styles.mobile} onClick={() => this.setSliderDrawer(true)}>
						<i className="fad fa-bars"/>
					</div>
					<div className="clear"/>
				</div>

				<div className={sliderClasses}>
					<div className={styles.background} onClick={() => this.setSliderDrawer(false)}/>
					<div className={styles.drawer}>
						<div className={styles.close} onClick={() => this.setSliderDrawer(false)}><i className="fad fa-times"/></div>
						<img src={MAIN_LOGO} alt=""/>
						<div className={styles.menu}>
							<Link to="/" onClick={() => this.setSliderDrawer(false)}>
								<div className={styles.menuItem}>
									<i className="fad fa-home-alt"/>
									HOME
								</div>
							</Link>
							<Link to="/spas-hot-tubs" onClick={() => this.setSliderDrawer(false)}>
								<div className={styles.menuItem}>
									<i className="fad fa-hot-tub"/>
									HOT TUBS & SPAS
								</div>
							</Link>
							<Link to="/swim-spas" onClick={() => this.setSliderDrawer(false)}>
								<div className={styles.menuItem}>
									<i className="fad fa-swimmer"/>
									SWIM SPAS
								</div>
							</Link>
							<Link to="/bbq-islands" onClick={() => this.setSliderDrawer(false)}>
								<div className={styles.menuItem}>
									<i className="fad fa-fire"/>
									BBQs
								</div>
							</Link>
							<Link to="/service-contact" onClick={() => this.setSliderDrawer(false)}>
								<div className={styles.menuItem}>
									<i className="fad fa-tools"/>
									SERVICE & CONTACT
								</div>
							</Link>
						</div>
						<div className={styles.callBlock}>
							<div className={styles.call}>
								<div className={styles.num}>(805) 473 - 2205</div>
								<div className={styles.label}>Call Arroyo Grande</div>
							</div>
							<div className={styles.call}>
								<div className={styles.num}>(805) 544 - 1772</div>
								<div className={styles.label}>Call San Luis Obispo</div>
							</div>
						</div>
					</div>
				</div>
			</>

		);
	}
}
