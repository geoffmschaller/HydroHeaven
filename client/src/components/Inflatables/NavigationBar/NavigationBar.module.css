.navigationBar {
  width: 100%;
  color: #f4f6f6;
  padding: 20px;
  z-index: 1;
  position: relative;
  background: #154360;
  background: -moz-linear-gradient(90deg, #154360 53%, #1f618d 100%);
  background: -webkit-linear-gradient(90deg, #154360 53%, #1f618d 100%);
  background: -webkit-gradient(linear, left top, right top, color-stop(53%, #154360), to(#1f618d));
  background: -webkit-linear-gradient(left, #154360 53%, #1f618d 100%);
  background: -o-linear-gradient(left, #154360 53%, #1f618d 100%);
  background: linear-gradient(90deg, #154360 53%, #1f618d 100%);
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#154360", endColorstr="#1f618d", GradientType=1); }
  .navigationBar .desktop {
    width: 80%;
    float: right;
    display: block; }
    .navigationBar .desktop a {
      color: #f4f6f6;
      text-decoration: none; }
    .navigationBar .desktop .menuItem {
      padding: 0 5px 5px 5px;
      margin: 0 20px;
      float: right;
      cursor: pointer;
      border-bottom: 2px solid transparent;
      -webkit-transition: all .3s ease-in-out;
      -o-transition: all .3s ease-in-out;
      transition: all .3s ease-in-out; }
      .navigationBar .desktop .menuItem:hover {
        border-bottom: 2px solid #f8c471; }
      .navigationBar .desktop .menuItem i {
        margin-right: 5px;
        font-size: 20px; }
  .navigationBar .mobile {
    width: 20%;
    float: left;
    display: none; }
    .navigationBar .mobile i {
      font-size: 30px;
      cursor: pointer; }

.slider {
  width: 100%;
  height: 100vh;
  -webkit-transition: all .2s ease-in-out;
  -o-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
  overflow: hidden;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: 1000; }
  .slider .background {
    position: relative;
    width: 100%;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.6);
    -webkit-transition: all .2s ease-in-out;
    -o-transition: all .2s ease-in-out;
    transition: all .2s ease-in-out;
    z-index: 1001; }
  .slider .drawer {
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    width: 90%;
    max-width: 400px;
    height: 100vh;
    background-color: white;
    padding: 10px;
    -webkit-transition: all .2s ease-in-out;
    -o-transition: all .2s ease-in-out;
    transition: all .2s ease-in-out;
    z-index: 1002; }
    .slider .drawer .close {
      width: 100%;
      font-size: 25px;
      text-align: right;
      cursor: pointer;
      margin-bottom: 20px; }
    .slider .drawer img {
      width: 100%;
      margin-bottom: 40px; }
    .slider .drawer .menu {
      margin-bottom: 30px; }
      .slider .drawer .menu a {
        color: #154360;
        text-decoration: none; }
      .slider .drawer .menu .menuItem {
        padding: 0 0 20px 5px; }
        .slider .drawer .menu .menuItem i {
          width: 25px; }
    .slider .drawer .callBlock {
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0; }
      .slider .drawer .callBlock .call {
        margin-bottom: 15px;
        text-align: center; }
        .slider .drawer .callBlock .call .num {
          font-size: 20px; }
        .slider .drawer .callBlock .call .label {
          font-size: 12px; }

.closedSlider {
  width: 0;
  height: 0;
  padding: 0;
  overflow: hidden; }

@media (max-width: 800px) {
  .navigationBar .desktop {
    display: none; }
  .navigationBar .mobile {
    display: block; } }

@media (min-width: 801px) and (max-width: 1050px) {
  .navigationBar .desktop {
    width: 100%; }
  .slider .background {
    width: 0;
    padding: 0;
    overflow: hidden; }
  .slider .drawer {
    width: 0;
    padding: 0;
    overflow: hidden; } }
