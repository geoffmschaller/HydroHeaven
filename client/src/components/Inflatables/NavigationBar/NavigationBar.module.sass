$bar-color: #154360
$text-color: #f4f6f6
$hover-border: #f8c471

.navigationBar
	width: 100%
	color: $text-color
	padding: 20px
	z-index: 1
	position: relative
	background: rgb(21, 67, 96)
	background: -moz-linear-gradient(90deg, rgba(21, 67, 96, 1) 53%, rgba(31, 97, 141, 1) 100%)
	background: -webkit-linear-gradient(90deg, rgba(21, 67, 96, 1) 53%, rgba(31, 97, 141, 1) 100%)
	background: linear-gradient(90deg, rgba(21, 67, 96, 1) 53%, rgba(31, 97, 141, 1) 100%)
	filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#154360", endColorstr="#1f618d", GradientType=1)

	.desktop
		width: 80%
		float: right
		display: block

		a
			color: $text-color
			text-decoration: none

		.menuItem
			padding: 0 5px 5px 5px
			margin: 0 20px
			float: right
			cursor: pointer
			border-bottom: 2px solid transparent
			transition: all .3s ease-in-out

			&:hover
				border-bottom: 2px solid $hover-border

			i
				margin-right: 5px
				font-size: 20px

	.mobile
		width: 20%
		float: left
		display: none

		i
			font-size: 30px
			cursor: pointer


.slider
	width: 100%
	height: 100vh
	transition: all .2s ease-in-out
	overflow: hidden
	position: absolute
	top: 0
	left: 0
	bottom: 0
	z-index: 1000

	.background
		position: relative
		width: 100%
		height: 100vh
		background-color: rgba(0, 0, 0, .6)
		transition: all .2s ease-in-out
		z-index: 1001

	.drawer
		position: absolute
		top: 0
		left: 0
		bottom: 0
		width: 90%
		max-width: 400px
		height: 100vh
		background-color: white
		padding: 10px
		transition: all .2s ease-in-out
		z-index: 1002

		.close
			width: 100%
			font-size: 25px
			text-align: right
			cursor: pointer
			margin-bottom: 20px

		img
			width: 100%
			margin-bottom: 40px

		.menu
			margin-bottom: 30px

			a
				color: $bar-color
				text-decoration: none

			.menuItem
				padding: 0 0 20px 5px

				i
					width: 25px

		.callBlock
			position: absolute
			bottom: 0
			left: 0
			right: 0

			.call
				margin-bottom: 15px
				text-align: center

				.num
					font-size: 20px

				.label
					font-size: 12px


.closedSlider
	width: 0
	height: 0
	padding: 0

@media(max-width: 800px)
	.navigationBar
		.desktop
			display: none

		.mobile
			display: block

@media(min-width: 801px) and (max-width: 1050px)
	.navigationBar
		.desktop
			width: 100%

	.slider
		.background
			width: 0
			padding: 0
			overflow: hidden

		.drawer
			width: 0
			padding: 0
			overflow: hidden