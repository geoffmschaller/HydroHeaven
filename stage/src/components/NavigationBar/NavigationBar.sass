$nav-background: #154360
$nav-highlight: #F5B041

#navigation-bar
	width: 100%
	position: relative

	#top-bar
		padding: 20px 10%

		#main-logo
			width: 100%
			max-width: 400px

		#call-block
			float: right
			padding-top: 20px

	#nav-bar
		width: 100%
		padding: 10px
		background-color: $nav-background
		position: relative

		#nav-items
			width: 100%
			max-width: 1000px
			margin: 0 auto
			color: white
			flex-direction: row
			justify-content: space-around
			text-transform: uppercase
			display: flex

			.item
				cursor: pointer
				border-bottom: 2px solid $nav-background
				transition: all .2s ease-in-out

				&:hover
					border-bottom: 2px solid $nav-highlight

	#mobile-menu-toggle
		padding: 0 20px
		font-size: 35px
		color: white
		display: none

	.mobile-menu
		width: 0
		background-color: white
		border: none
		height: 100vh
		position: absolute
		top: 0
		left: 0
		bottom: 0
		z-index: 10000
		padding: 0
		transition: all .2s ease-in-out
		overflow-x: hidden

		#close
			width: 100%
			text-align: right
			font-size: 30px
			margin-bottom: 30px
			color: #D5D8DC

		.item
			width: 100%
			font-size: 18px
			margin-bottom: 20px


	.open
		width: 80%
		max-width: 400px
		padding: 50px 20px
		border-right: 5px solid $nav-background
		overflow: hidden

@media (max-width: 850px)
	#navigation-bar
		#top-bar
			#main-logo
				width: 100%
				max-width: 100%

			#call-block
				width: 100%
				text-align: center

		#nav-bar
			#nav-items
				display: none

		#mobile-menu-toggle
			display: block

		#mobile-menu
			display: block