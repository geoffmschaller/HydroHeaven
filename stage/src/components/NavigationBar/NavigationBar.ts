import {Component, Vue} from 'vue-property-decorator';

@Component
export default class NavigationBar extends Vue {

	mobileIsOpen: boolean = false;
	topOffset: number = 0;

	toggleMobileMenu(mode: boolean) {
		if (mode) {
			this.topOffset = window.scrollY;
			document.body.style.overflow = 'hidden';
		} else {
			document.body.style.overflow = 'auto';
		}
		this.mobileIsOpen = mode;
	}

}