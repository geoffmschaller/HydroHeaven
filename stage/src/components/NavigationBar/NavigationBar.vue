<template>
	<div id="navigation-bar">
		<div id="top-bar">
			<img id="main-logo" :src="require('@/assets/images/main_logo_desktop_shaddow.png')" alt="">
			<div id="call-block">
				<div>Arroyo Grande: (805) 473 - 2205</div>
				<div>San Luis Obispo: (805) 544 - 1772</div>
			</div>
			<div class="clear"/>
		</div>
		<div id="nav-bar">
			<div id="mobile-menu-toggle" @click="toggleMobileMenu(true)"><i class="fas fa-bars"/></div>
			<div id="nav-items">
				<div class="item">Home</div>
				<div class="item">Hot Tubs & Spas</div>
				<div class="item">Swim Spas</div>
				<div class="item">BBQs & Islands</div>
				<div class="item">Service & Contact</div>
			</div>
		</div>
		<div class="mobile-menu" v-bind:class="{open: mobileIsOpen}" v-bind:style="{top: `${topOffset}px`}">
			<div id="close" @click="toggleMobileMenu(false)"><i class="fas fa-times"/></div>
			<div class="item">Home</div>
			<div class="item">Hot Tubs & Spas</div>
			<div class="item">Swim Spas</div>
			<div class="item">BBQs & Islands</div>
			<div class="item">Service & Contact</div>
		</div>
	</div>
</template>

<script src="./NavigationBar.ts"/>
<style scoped lang="sass" src="./NavigationBar.sass"/>