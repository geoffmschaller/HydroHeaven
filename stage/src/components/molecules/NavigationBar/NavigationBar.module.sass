@import '../../../themes/Main'

.navigationBar
	width: 100%
	padding: 2rem
	position: fixed
	top: 0
	left: 0
	right: 0
	z-index: 100000

	.socialBar
		display: flex
		flex-direction: row
		justify-content: flex-end
		align-items: center
		color: white

		a
			color: white
			text-decoration: none

			i
				font-size: 1.2rem
				margin: 0 0.7rem

		.phone
			font-size: 1.2rem
			margin: 0 0.7rem

	.menuBar
		display: flex
		flex-direction: row
		justify-content: space-between
		align-content: center
		width: 100%
		max-width: 1400px
		margin: 0 auto
		margin-top: 2rem

		.logo
			display: flex
			flex-direction: row
			align-content: center
			color: white

			.companyName
				font-size: 1.6rem

		.menu
			display: flex
			flex-direction: row
			justify-content: flex-end
			align-content: center
			color: white
			font-size: 1rem
			overflow: hidden
			height: 2rem

			a
				text-decoration: none
				color: inherit

			.item
				margin: 0 1rem
				cursor: pointer
				transition: all .2s ease-in-out

				&:hover
					color: $accent

			i
				font-size: 2rem
				cursor: pointer

	.mobileMenu
		display: none
		position: absolute
		top: 0
		left: 0
		bottom: 0
		width: 0
		transition: all .2s ease-in-out
		background-color: $dark
		padding-top: 5rem

		.item
			margin: 0 1rem
			cursor: pointer
			padding: 2rem 0
			transition: all .2s ease-in-out
			color: white

			&:hover
				color: $accent

			i
				font-size: 2rem
				cursor: pointer

	.openMenu
		display: block
		width: 15rem
		height: 100vh

.home
	background-color: transparent

.hotTubs
	background-color: $dark

