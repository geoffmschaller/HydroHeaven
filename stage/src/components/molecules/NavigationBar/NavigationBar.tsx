import React, {FunctionComponent, useState, useEffect} from 'react';
import {Link} from 'react-router-dom';
import styles from './NavigationBar.module.sass';
import MAIN_THEME from '../../../themes/Main';
import SlideDown from '../../../animations/SlideDown';
import SlideUp from "../../../animations/SlideUp";

interface Props {
	background: String
}

const NavigationBar: FunctionComponent<Props> = (props) => {

	const [currentWidth, setCurrentWidth] = useState(0);
	const [isMenuOpen, setIsMenuOpen] = useState(false);

	useEffect(() => {
		setCurrentWidth(window.innerWidth)
	}, [])

	const setMenu = (mode: boolean = !isMenuOpen) => {
		setIsMenuOpen(mode);
	}

	let backgroundColor;
	switch (props.background) {
		case "home":
			backgroundColor = 'transparent';
			break;
		default:
			backgroundColor = MAIN_THEME.dark;
	}

	return (
		<div className={styles.navigationBar} style={{backgroundColor: backgroundColor}}>
			<div className={styles.socialBar}>
				<SlideDown>
					<a href={'https://www.facebook.com/pages/category/Health-Spa/Hydro-Heaven-831250550309179/'} target={'_blank'} rel="noreferrer">
						<i className="fab fa-facebook"/>
					</a>
				</SlideDown>
				<SlideDown duration={0.3}>
					<a href={'https://www.yelp.com/biz/hydro-heaven-spas-stoves-and-bbqs-arroyo-grande'} target={'_blank'} rel="noreferrer">
						<i className="fab fa-yelp"/>
					</a>
				</SlideDown>
				<SlideDown duration={0.5}>
					<div className={styles.phone}>(805) 473 - 2205</div>
				</SlideDown>
			</div>
			<div className={styles.menuBar}>
				<div className={styles.logo}>
					<div className={styles.companyName}>Hydro Heaven Spas</div>
				</div>
				<div className={styles.menu}>
					{
						currentWidth <= 900
							? <i className="fas fa-bars" onClick={() => setMenu()}/>
							: <>
								<SlideUp>
									<Link to={'/'}>
										<div className={styles.item}>Home</div>
									</Link>
								</SlideUp>
								<SlideUp duration={0.3}>
									<Link to={'/spas-hot-tubs'}>
										<div className={styles.item}>Hot Tubs</div>
									</Link>
								</SlideUp>
								<SlideUp duration={0.4}>
									<Link to={'/swim-spas'}>
										<div className={styles.item}>Swim Spas</div>
									</Link>
								</SlideUp>
								<SlideUp duration={0.5}>
									<Link to={'/bbq-islands'}>
										<div className={styles.item}>BBQs</div>
									</Link>
								</SlideUp>
								<SlideUp duration={0.6}>
									<Link to={'/service-contact'}>
										<div className={styles.item}>Service/Contact</div>
									</Link>
								</SlideUp>
							</>
					}
				</div>
			</div>
			<div className={[styles.mobileMenu, isMenuOpen ? styles.openMenu : ""].join(" ")}>
				<Link to={'/'}>
					<div className={styles.item}>Home</div>
				</Link>
				<Link to={'/spas-hot-tubs'}>
					<div className={styles.item}>Hot Tubs</div>
				</Link>
				<Link to={'/swim-spas'}>
					<div className={styles.item}>Swim Spas</div>
				</Link>
				<Link to={'/bbq-islands'}>
					<div className={styles.item}>BBQs</div>
				</Link>
				<Link to={'/service-contact'}>
					<div className={styles.item}>Service/Contact</div>
				</Link>
			</div>
		</div>
	)
}

export default NavigationBar;