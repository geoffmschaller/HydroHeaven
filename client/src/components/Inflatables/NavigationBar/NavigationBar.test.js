import React from 'react';
import NavigationBar from "./NavigationBar";

const wrapper = <NavigationBar/>;

describe("DashboardNavigation Bar Inflatable", () => {

	it("Renders Correctly", () => {
		expect(wrapper).toMatchSnapshot();
	});

});
