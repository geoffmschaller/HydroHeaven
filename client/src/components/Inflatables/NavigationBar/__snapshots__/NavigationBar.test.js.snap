// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardNavigation Bar Inflatable 1`] = `<NavigationBar />`;
